homeassistant/components/dsmr_reader/__init__.py

```diff
@@ -1 +1,15 @@
 """The DSMR Reader component."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the DSMR Reader integration."""
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "sensor")
+    )
+    return True
```

Comment:
Missing `async_unload_entry`

Reply:
Ah, that explains why HA was saying it needed to be restarted for the changes to take effect. I've taken the time to also simplify the `async_setup_entry` as it was just doing the work that `async_setup_platforms()` already does for integrations.